tests/dash/test_dash_vnet.py

```diff
@@ -0,0 +1,241 @@
+        self.overlay_ip = "10.0.0.6"
+        pb = Route()
+        pb.action_type = RoutingType.ROUTING_TYPE_VNET_DIRECT
+        pb.vnet_direct.vnet = self.vnet
+        pb.vnet_direct.overlay_ip.ipv4 = socket.htonl(int(ipaddress.ip_address(self.overlay_ip)))
+        dash_db.create_route(self.group_id, self.ip, {"pb": pb.SerializeToString()})
+        time.sleep(3)
+        outbound_routing_entries = dash_db.asic_outbound_routing_table.get_keys()
+        assert outbound_routing_entries
+        fvs = dash_db.asic_outbound_routing_table[outbound_routing_entries[0]]
+        for fv in fvs.items():
+            if fv[0] == "SAI_OUTBOUND_ROUTING_ENTRY_ATTR_ACTION":
+                assert fv[1] == "SAI_OUTBOUND_ROUTING_ENTRY_ACTION_ROUTE_VNET_DIRECT"
+            if fv[0] == "SAI_OUTBOUND_ROUTING_ENTRY_ATTR_OVERLAY_IP":
+                assert fv[1] == "10.0.0.6"
+        assert "SAI_OUTBOUND_ROUTING_ENTRY_ATTR_DST_VNET_ID" in fvs
+
+        pb = EniRoute()
+        pb.group_id = self.group_id
+        self.mac_string = "F4939FEFC47E"
+        dash_db.create_eni_route(self.mac_string, {"pb": pb.SerializeToString()})
+        time.sleep(3)
+        eni_entries = dash_db.asic_eni_table.get_keys()
+        fvs = dash_db.asic_eni_table[eni_entries[0]]
+        for fv in fvs.items():
+            if fv[0] == "SAI_ENI_ATTR_OUTBOUND_ROUTING_GROUP_ID":
+                assert fv[1] == outbound_routing_group_entries[0]
+
+    def test_inbound_routing(self, dash_db):
+        self.mac_string = "F4939FEFC47E"
+        self.vnet = "Vnet1"
+        self.vni = "3251"
+        self.ip = "10.1.1.1"
+        self.action_type = "decap"
+        self.pa_validation = "true"
+        self.priority = "1"
+        self.protocol = "0"
+        pb = RouteRule()
+        pb.pa_validation = True
+        pb.priority = int(self.priority)
+        pb.protocol = int(self.protocol)
+        pb.vnet = self.vnet
+
+        dash_db.create_inbound_routing(self.mac_string, self.vni, self.ip, {"pb": pb.SerializeToString()})
+        time.sleep(3)
+
+        inbound_routing_entries = dash_db.asic_inbound_routing_rule_table.get_keys()
+        assert inbound_routing_entries
+        fvs = dash_db.asic_inbound_routing_rule_table[inbound_routing_entries[0]]
+        for fv in fvs.items():
+            if fv[0] == "SAI_INBOUND_ROUTING_ENTRY_ATTR_ACTION":
+                assert fv[1] == "SAI_INBOUND_ROUTING_ENTRY_ACTION_TUNNEL_DECAP_PA_VALIDATE"
+
+    def test_cleanup(self, dash_db):
+        self.vnet = "Vnet1"
+        self.mac_string = "F4939FEFC47E"
+        self.group_id = ROUTE_GROUP1
+        self.vni = "3251"
+        self.sip = "10.1.1.1"
+        self.dip = "10.1.0.0/24"
+        self.ip2 = "10.1.1.2"
+        self.appliance_id = "100"
+        self.routing_type = "vnet_encap"
+        dash_db.remove_inbound_routing(self.mac_string, self.vni, self.sip)
+        dash_db.remove_eni_route(self.mac_string)
+        dash_db.remove_route(self.group_id, self.dip)
```

Comment:
do we need remove_route after unbind ?

Reply:
yes, due to current limitations in the VS SAI. Once VS SAI is updated I'll come back and revisit these tests.